### PR TITLE
Enable unused variables warning

### PR DESCRIPTION
**Pull request by rachitnigam**
@tedbauer I'm enabling stricter checks for the project. The build system throws a few unused variable warnings now. Can you build this branch and change any intentional unused variables to `_`?

### CODE REVIEW

**Comment by tedbauer** [comment]
Ok, I believe I've just changed all the unused variables to "_". There are also a lot of non-exhaustive pattern matching, would you like me to fix that?

File: src/emit.ml [hunk 1]
```diff
@@ -141,12 +141,17 @@ let rec emit_cmd i cmd =
...
 and emit_assign_int (id, e) =
   concat [ "int "; id; " = "; (emit_expr e); ";" ]
 
+<<<<<<< HEAD
 and emit_assign_arr (id, _, d) i =
+=======
+(* FIXME(rachit): [e] is unused. Remove it from param list or use it. *)
+and emit_assign_arr (id, e, d) i =
+>>>>>>> c1b794ad77d439e39b941fb48e3fd637ca37f61d
   let bf = compute_bf d in
```
Comment (rachitnigam) [inline src/emit.ml]: ^ Fix please.

File: src/emit.ml [hunk 2]
```diff
@@ -215,7 +220,11 @@ and emit_typedef (id, t) i =
   concat [ "typedef "; (type_str t); " "; id; ";" ]
   |> indent i
 
+<<<<<<< HEAD
 and emit_muxdef _ = "" (* No need to emit anything *)
+=======
+and emit_muxdef = "" (* No need to emit anything *)
+>>>>>>> c1b794ad77d439e39b941fb48e3fd637ca37f61d
```
Comment (rachitnigam) [inline src/emit.ml]: Fix please

File: src/type.ml [hunk 3]
```diff
@@ -145,20 +142,21 @@ and check_for_impl id r1 r2 body u (ctx, delta) =
     else raise (TypeError range_static_error)
   | _ -> raise (TypeError range_error)
 
+(* TODO(rachit): [d] is unused. Is this a mistake? *)
 and check_assignment id exp (ctx, delta) =
-  check_expr exp (ctx, delta) |> fun (t, (c, d)) ->
```
Comment (rachitnigam) [inline src/type.ml]: Are you sure this is correct? If so, can you add a comment explaining why?

File: src/type.ml [hunk 4]
```diff
@@ -127,10 +124,10 @@ and check_for id r1 r2 body (ctx, d) =
   | _ -> raise (TypeError range_error)
 
 and check_for_impl id r1 r2 body u (ctx, delta) =
-  check_expr r1 (ctx, delta) |> fun (r1_type, (c1, d1)) ->
```
Comment (rachitnigam) [inline src/type.ml]: Are you sure this is correct? The definition for `check_for` has different logic.

**Comment by rachitnigam** [comment]
Yeah, might as well. You can just add cases saying `failwith "Not yet implemented"`.

**Pull request merged**